src/clients/QBittorrent.ts

```diff
@@ -244,39 +240,53 @@ export default class QBittorrent implements TorrentClient {
 	 */
 	async generateCorrectSavePath(
 		searchee: Searchee,
-		{ save_path },
-	): Promise<{ save_path: string }> {
-		const subfolderContentLayout =
-			await this.isSubfolderContentLayout(searchee);
+		torrentInfo: TorrentInfo,
+	): Promise<string> {
+		const subfolderContentLayout = await this.isSubfolderContentLayout(
+			searchee,
+			torrentInfo,
+		);
 		if (subfolderContentLayout) {
-			return {
-				save_path: join(save_path, stripExtension(searchee.name)),
-			};
+			return dirname(torrentInfo.content_path);
 		}
-		return { save_path };
+		return torrentInfo.save_path;
+	}
+
+	/*
+	@return array of query results
+	 */
+	async getAllTorrentInfo(): Promise<TorrentInfo[]> {
```

Comment:
Can we leverage some of this in the future for #471 ?

The function names seem to imply some of that functionality would be available?

Reply:
This is calling the same /torrents/info endpoint but it gets all torrents in client. It was done this way as since we are only calculating v1 hash, if a torrent has v2 it won't return anything. So you get all torrents and check all the different hashes on it for a match.

As for that feature. I can look into it.